rclcpp/src/rclcpp/experimental/executors/events_executor/events_executor.cpp

```diff
@@ -341,6 +341,11 @@ EventsExecutor::refresh_current_collection(
   // Acquire lock before modifying the current collection
   std::lock_guard<std::mutex> guard(mutex_);
 
+  // Remove expired entities to ensure re-initialized objects
+  // are updated. This fixes issues with stale state entities.
+  // See: https://github.com/ros2/rclcpp/pull/2586
+  current_collection_.remove_expired_entities();
```

Comment:
It looks like this fix is also applied to the `Executor` class, so it's correct to have it also here.

Reply:
Just to clarify, is there an action required from this comment?
I can think of adding a link to the original issue where the bug was first reported: https://github.com/ros2/rclcpp/issues/2180

Reply:
Nothing to do